src/python/pants/core/goals/test.py

```diff
@@ -80,22 +93,39 @@ class TestResult(EngineAwareReturnType):
     # Prevent this class from being detected by pytest as a test class.
     __test__ = False
 
-    @classmethod
-    def skip(cls, address: Address, output_setting: ShowOutput) -> TestResult:
-        return cls(
+    @staticmethod
+    def skip(address: Address, output_setting: ShowOutput) -> TestResult:
+        return TestResult(
             exit_code=None,
             stdout="",
             stderr="",
             stdout_digest=EMPTY_FILE_DIGEST,
             stderr_digest=EMPTY_FILE_DIGEST,
-            address=address,
+            addresses=(address,),
             output_setting=output_setting,
             result_metadata=None,
         )
 
-    @classmethod
+    @staticmethod
+    def skip_batch(
+        batch: TestRequest.Batch[_TestFieldSetT, Any], output_setting: ShowOutput
+    ) -> TestResult:
+        return TestResult(
+            exit_code=None,
+            stdout="",
+            stderr="",
+            stdout_digest=EMPTY_FILE_DIGEST,
+            stderr_digest=EMPTY_FILE_DIGEST,
+            addresses=tuple(field_set.address for field_set in batch.elements),
+            output_setting=output_setting,
+            result_metadata=None,
+            partition_description=batch.partition_metadata.description
+            if batch.partition_metadata
+            else None,
```

Comment:
I think if you parenthesize this, black will keep it and make it more legible

Reply:
Indeed. Not necessary, but nice to do